libsolidity/analysis/TypeChecker.cpp

```diff
@@ -737,6 +737,46 @@ void TypeChecker::endVisit(ExpressionStatement const& _statement)
 			typeError(_statement.expression().location(), "Invalid integer constant.");
 }
 
+bool TypeChecker::visit(Conditional const& _conditional)
+{
+	expectType(_conditional.condition(), BoolType());
+
+	_conditional.trueExpression().accept(*this);
+	_conditional.falseExpression().accept(*this);
+
+	TypePointer const& trueType = type(_conditional.trueExpression());
+	TypePointer const& falseType = type(_conditional.falseExpression());
+
+	TypePointer commonType = Type::commonType(trueType, falseType);
+	if (!commonType)
+		// we fake it as an equal operator, but any other comparison operator can work.
+		commonType = trueType->binaryOperatorResult(Token::Equal, falseType);
```

Comment:
``Type::commonType()`` and ``binaryOperatorResult()``, maybe we can just add another check for integer const type after ``Type::commonType()``, but I think reusing the existing function is more clear.